### PR TITLE
Face Recognition

### LINKED ISSUE #456: Face Detection with OpenCV's built-in recognizer
Aim: This script recognizes the face in an image using OpenCV's built-in recognizer. 
Feature: Easy and understandable code 

I would like to work on this. Kindly assign me.

### PR DESCRIPTION
**Pull request by aakriti1318**
# Description

Please include a summary of the change and which issue is fixed. List any dependencies that are required for this change.

## Fixes #456 

Replace `issue_no` with the issue number which is fixed in this PR

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- Yes

## Type of change

_Please delete options that are not relevant._

-  Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update
- Documentation Update

## Checklist:

-  My code follows the style guidelines(Clean Code) of this project
-  I have performed a self-review of my own code
-  I have commented my code, particularly in hard-to-understand areas
-  I have created a helpful and easy to understand `README.md`
-  My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
-  My changes generate no new warnings
-  I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

File: Face-Detection/face_train.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Face Recognition with OpenCV built in classifier
```
Comment (kaustubhgupta) [inline Face-Detection/face_train.py:1]: Delete this file

File: Face-Detection/Recognition.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import numpy as np
```
Comment (kaustubhgupta) [inline Face-Detection/Recognition.py:1]: Delete this file

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
After looking at Pylint logs, here is what I think needs to be deleted

File: Face_Recognition/face_train.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+for i in os.listdir(DIR):  # Will append the names of the person in the people list
+    people.append(i)
+
+haar_cascade = cv.CascadeClassifier('haar_face.xml')
+
+# For every face in the list the corresponding labels to the features are designated.
+features = []
+labels = []
+
+def train_face(features):
```
Comment (antrikshmisri) [inline Face_Recognition/face_train.py:21]:
```suggestion
def train_face(features:list):
```

Comment (antrikshmisri) [reply Face_Recognition/face_train.py:21]: @aakriti1318 As I mentioned earlier , the issue is that the features argument is being treated as a ndarray and you are trying to call append function which is causing the error. So just explicitly define its type as list

Comment (kaustubhgupta) [reply Face_Recognition/face_train.py:21]: Yes, that's the issue, the features and labels are called twice once as a list and then as ndarray. That's why Pylint is trying to refer it as ndarray but when it gets append member it causes error

Comment (antrikshmisri) [reply Face_Recognition/face_train.py:21]: @kaustubhgupta Exactly

**Comment by kaustubhgupta** [comment]
@aakriti1318 updates on this?

File: Face_Recognition/face_train.py (hunk 3 above)
Comment (aakriti1318) [reply Face_Recognition/face_train.py]: I have changed my entire code still not working.

**Comment by kaustubhgupta** [comment]
@aakriti1318 You are not reading our suggestions properly, We suggested deleting 2 files and shared what the Pylint has returned. You should check the Lint Code Base logs to get information about where your code has errors

File: Face_Recognition/face_detection.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+face_cascade = cv2.CascadeClassifier("haarcascade_frontalface_alt.xml")  # classifier
+while True:
+	ret, frame = cap.read()    # we read 1 img 
+	gray_frame = cv2.cvtColor(frame,cv2.COLOR_BGR2GRAY)
+	if ret==False:
+		continue
+# now we read the img
+										# (gray_frame, scaling factor, no of neighbors)
+	faces = face_cascade.detectMultiscale(gray_frame,1.3,5)   # 1.3 means img will shrink by 30% 
+ 	cv2.imshow("Video Frame",frame)
```
Comment (kaustubhgupta) [inline Face_Recognition/face_detection.py:14]: Check this error. I think you have to remove all the space before cv2 and use one tab
```
Face_Recognition/face_detection.py:14:0: E0001: inconsistent use of tabs and spaces in indentation (<unknown>, line 14) (syntax-error)
```

File: Face_Recognition/face_recognition.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+		break
+
+cap.release()
+cv2.destroyAllWindows()
+
+import numpy as np 
+import cv2
+import os 
+
+def distance(v1, v2):
```
Comment (kaustubhgupta) [inline Face_Recognition/face_recognition.py:124]: This function is already defined in line 14. So remove this

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
@aakriti1318 Implement these

**Comment by kaustubhgupta** [comment]
@aakriti1318 Even after multiple reminders, suggestions, there was no response from your side and that's why this issue and PR have been closed.

**Pull request closed without merging**